x/restaking/simulation/msg_factory.go

```diff
@@ -0,0 +1,613 @@
+	}
+}
+
+func SimulateMsgJoinService(
+	ak authkeeper.AccountKeeper,
+	bk bankkeeper.Keeper,
+	opk types.OperatorsKeeper,
+	sk types.ServicesKeeper,
+) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		msg := &types.MsgJoinService{}
+
+		operatorsKeeper := opk.(*operatorskeeper.Keeper)
+		servicesKeeper := sk.(*serviceskeeper.Keeper)
+
+		// No account skipping
+		if len(accs) == 0 {
+			return simtypes.NoOpMsg(types.ModuleName, sdk.MsgTypeURL(msg), "no accounts"), nil, nil
+		}
+
+		// Get a random operator
+		operator, found := operatorssimulation.GetRandomExistingOperator(r, ctx, operatorsKeeper, nil)
+		if !found {
+			return simtypes.NoOpMsg(types.ModuleName, sdk.MsgTypeURL(msg), "could not get operator"), nil, nil
+		}
+
+		// Get a random service
+		service, found := servicessimulation.GetRandomExistingService(r, ctx, servicesKeeper, nil)
+		if !found {
+			return simtypes.NoOpMsg(types.ModuleName, sdk.MsgTypeURL(msg), "could not get service"), nil, nil
+		}
+
+		msg = types.NewMsgJoinService(operator.ID, service.ID, operator.Admin)
+
+		// Get the admin account that should sign the transaction
+		adminAddress, err := sdk.AccAddressFromBech32(operator.Admin)
+		if err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, sdk.MsgTypeURL(msg), "error while parsing admin address"), nil, nil
+		}
+		signer, found := simtesting.GetSimAccount(adminAddress, accs)
+		if !found {
+			return simtypes.NoOpMsg(types.ModuleName, sdk.MsgTypeURL(msg), "admin account not found"), nil, nil
+		}
+
+		return simtesting.SendMsg(r, types.ModuleName, app, ak, bk, msg, ctx, signer)
+	}
+}
+
+func SimulateMsgLeaveService(
+	ak authkeeper.AccountKeeper,
+	bk bankkeeper.Keeper,
+	opk types.OperatorsKeeper,
+	sk types.ServicesKeeper,
+	k *keeper.Keeper,
+) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		msg := &types.MsgLeaveService{}
+
+		operatorsKeeper := opk.(*operatorskeeper.Keeper)
+
+		// No account skipping
+		if len(accs) == 0 {
+			return simtypes.NoOpMsg(types.ModuleName, sdk.MsgTypeURL(msg), "no accounts"), nil, nil
+		}
+
+		// Get all the services
+		services, err := sk.GetServices(ctx)
+		if err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, sdk.MsgTypeURL(msg), "could not get services"), nil, nil
+		}
+
+		// Get a operator that has joined a service
+		var service servicestypes.Service
+		operator, found := operatorssimulation.GetRandomExistingOperator(r, ctx, operatorsKeeper, func(o operatorstypes.Operator) bool {
+			// Search a service that the operator has joined
+			for _, s := range services {
+				hasJoined, _ := k.HasOperatorJoinedService(ctx, o.ID, s.ID)
```

Comment:
Instead of `o` and `s` I would suggest using names as `operator` and `service` to make it easier to read